### DISCUSSION TITLE
Pooling together top Apple developer tooling issues

### DISCUSSION THREAD

**Discussion by saagarjha** [MobileNativeFoundation/discussions#204] [Proposals]
Most (I would say "all", but I know some of you have really wild build systems) of Mobile Native Foundation depends heavily on Apple's developer tools when building software for their platforms. This shared tooling can be incredibly helpful when it allows us to share effort and be on the same page. However, this also means that bugs in this area can affect many of us. At scale, this has a tremendous cost, likely in the many millions of dollars.

Apple, of course, works hard to fix bugs in their software, and they take feedback from us when deciding when to fix. However, the process of resolving these is almost solely in their hands, and the official channels for providing input (feedbacks, devrel) are fragmented and isolating. In practice, what often ends up happening is engineers collaborate and share information about bugs and workarounds in an ad-hoc fashion to get around this. This results in wasted effort across companies (at least, until they find out about these discussions), but it also misrepresents issue severity to Apple: they might see 1-2 reports for an issue that affects fleets representing thousands of engineers. Frequently, this means that serious issues do not get fixed or even looked at for several releases. Apple engineers, to their credit, do take input from nontraditional channels, such as social media or relationships with individual developers, but this obviously does not scale nor does it fix the problem of representing how widespread an issue is.

I think it would greatly helpful if Mobile Native Foundation created a shared page with these issues, who is experiencing them, and work done to analyze and mitigate them. This would pool knowledge that used to be spread across social media/Slack/private issue trackers, but more importantly allow for a place for tracking widespread problems that aren't being addressed. I cannot speak for how this might affect fix timelines but I am sure that "this regression has not been fixed for the last 6 months and all these companies consider it a top issue" would be a much stronger driver than "some dude on Twitter complained about this" for prioritizing fixes when shared with Apple leadership.

**Comment by keith** [comment]
Hey @saagarjha, we chatted about this right before the new year in the MNF and I think we're pretty interested in this idea as well. We're starting out by reaching out to the openradar maintainer to see if we could potentially work with what already exists

**Reply by saagarjha** [reply]
Sounds great! Would be interested to hear how that discussion goes :)

**Reply by timburks** [reply]
Hi, I'm the originator and maintainer of Open Radar.  As discussed today on a call, we've agreed to move Open Radar to the MNF as a git repo containing a YAML+Markdown export of the full issue+comment history with an associated GitHub Issues section for new submissions. The repo will be an archive and the issues area would be used for all new submissions. There are no plans to sync the exported archive and the new issues, but that could be done as a followup. When this is done, the existing site will be replaced with a redirect to the new repo.

@tyvsmith @ob + others please followup with corrections or revisions

**Reply by saagarjha** [reply]
Awesome! Was there any discussion of moderation, marking issues as duplicates, etc.?
